### PR TITLE
feat(JobCard): remove onShelfSectionLinkClick and add linkType and payload to LinkComponent

### PR DESCRIPTION
**Pull request by cm-seekasia**
Related to https://github.com/seekinternational/seek-asia-style-guide/pull/304
Trello card: https://trello.com/c/GV9tkv6q

Need to set omniture tracking on JobCard. Id and payload are added to LinkComponent.

## How to Test:
1. Go to JobCard
2. Check Custom Link, Job Title Component
3. Click JobTitle, get `jobTitleLink` from console.
4. Click CompanyLink, get `companyLink` from console.
5. Click Location, get `locationLink`
6. Check shelf links, click more, click job function link, get linkType and payload

### CODE REVIEW

File: react/JobCard/components/JobTitleLink/JobTitleLink.js [hunk 1]
```diff
@@ -35,7 +35,7 @@ const JobTitleLink = ({ keyword, job: { jobTitle, jobUrl }, LinkComponent = defa
     );
   }
 
-  return <LinkComponent link={jobUrl} className={classnames({ [styles.jobTitleLink]: !viewed, [styles.jobTitleLinkVisited]: viewed })} rel="noopener noreferrer">{title}</LinkComponent>;
+  return <LinkComponent id="JobTitleLink" link={jobUrl} className={classnames({ [styles.jobTitleLink]: !viewed, [styles.jobTitleLinkVisited]: viewed })} rel="noopener noreferrer">{title}</LinkComponent>;
```
Comment (kosanna) [inline react/JobCard/components/JobTitleLink/JobTitleLink.js]: id's and data attributes usually follow `kebab-case`. So it would be good to have them consistent as eg `job-title-link`

File: react/JobCard/components/Link/Link.js [hunk 2]
```diff
@@ -1,12 +1,14 @@
 import React from 'react';
 import PropTypes from 'prop-types';
 
-const Link = ({ link, children, ...restProps }) => {
-  return (<a href={link} {...restProps}>{children}</a>);
+const Link = ({ link, children, id, payload, ...restProps }) => {
+  return (<a href={link} onClick={() => console.log('id: ', id, 'payload: ', payload)} {...restProps}>{children}</a>);
```
Comment (kosanna) [inline react/JobCard/components/Link/Link.js]: I am not convinced component should be doing console.logs 🤔 Why?

Comment (cm-seekasia) [reply react/JobCard/components/Link/Link.js]: This is a default link component, which console is for demo. 
Custom link component can do what ever they want based on id and payload.

**Comment by cm-seekasia** [comment]
> Why it's been decided to change the approach back from passing the click handler?

Cos there are many areas need to be tracked.

**Comment by kosanna** [comment]
Could you please provide an example usage on how it will be used? Right now component just preforms console.log 🤔 where payload will be coming from?

File: react/JobCard/JobCard.demo.js [hunk 3]
```diff
@@ -195,21 +192,25 @@ export default {
                     searchMethod: 'Browse Job Function - SRP',
                     items: [
                       {
+                        id: '0',
```
Comment (ChanJit) [inline react/JobCard/JobCard.demo.js]: any reason why adding the id in the demo? cause it looks like not related to this PR

File: react/JobCard/components/CompanyLink/CompanyLink.js [hunk 4]
```diff
@@ -33,7 +33,7 @@ const CompanyLink = ({ company, keyword = '', LinkComponent = defaultLink }) =>
   </span>);
 
   if (link) {
-    return (<LinkComponent link={link} className={styles.companyLink} title={title}>{companyText}</LinkComponent>);
+    return (<LinkComponent link={link} className={styles.companyLink} title={title} id="company-link">{companyText}</LinkComponent>);
```
Comment (ChanJit) [inline react/JobCard/components/CompanyLink/CompanyLink.js]: okay i get it now.

File: react/JobCard/JobCard.demo.js (hunk 3 above)
Comment (cm-seekasia) [reply react/JobCard/JobCard.demo.js]: Just to demo it works. Click on the link can get the id.

File: react/JobCard/components/Link/Link.js (hunk 2 above)
Comment (ChanJit) [reply react/JobCard/components/Link/Link.js]: is should be remove.

File: react/JobCard/JobCard.demo.js (hunk 3 above)
Comment (cm-seekasia) [reply react/JobCard/JobCard.demo.js]: It can be null/undefined if individual/micro tracking is not needed.

File: react/JobCard/components/Link/Link.js (hunk 2 above)
Comment (cm-seekasia) [reply react/JobCard/components/Link/Link.js]: It is for demo. Any suggestion on how to demo it?

Comment (cm-seekasia) [reply react/JobCard/components/Link/Link.js]: Let's move to demo.js only =]

**Comment by cm-seekasia** [comment]
Let's change id to linkType and move the console.log to demo only =]

File: react/JobCard/components/Link/Link.js [hunk 5]
```diff
@@ -8,5 +8,7 @@ const Link = ({ link, children, ...restProps }) => {
 export default Link;
 Link.propTypes = {
   link: PropTypes.string,
+  linkType: PropTypes.string,
+  payload: PropTypes.object,
```
Comment (ChanJit) [inline react/JobCard/components/Link/Link.js]: can you make the `link` and `children` as `isRequired` as well?

File: react/JobCard/components/Link/Link.js (hunk 2 above)
Comment (kosanna) [reply react/JobCard/components/Link/Link.js]: If it is for demo it should be in the demo only 👍 @ChanJit made an 100% valid point, that if default component is used then onClick will be overridden. I really prefer the previous solution more when we pass a click handler in, but I do agree it should live in the shared Link component. Thoughts? Could have a call once we are all free.

File: react/JobCard/components/Link/Link.js (hunk 5 above)
Comment (kosanna) [reply react/JobCard/components/Link/Link.js]: children are required, but link is not, because it's defaulted to default link component.

File: react/JobCard/components/CompanyLink/CompanyLink.js [hunk 6]
```diff
@@ -33,7 +33,7 @@ const CompanyLink = ({ company, keyword = '', LinkComponent = defaultLink }) =>
   </span>);
 
   if (link) {
-    return (<LinkComponent link={link} className={styles.companyLink} title={title}>{companyText}</LinkComponent>);
+    return (<LinkComponent link={link} className={styles.companyLink} title={title} linkType="companyLink">{companyText}</LinkComponent>);
```
[the remark this answers is not in the document]
Comment (kosanna) [reply react/JobCard/components/CompanyLink/CompanyLink.js]: Yup, we've introduced a bunch of console warnings with that. 
If we are going down that pass we could name it something like `data-link-type`.
https://reactjs.org/blog/2017/09/08/dom-attributes-in-react-16.html

**Comment by kosanna** [comment]
The api is not great because you need to remember that you need to pass all those things into LinkComponent thus it's not tracked automatically. What do you think of passing onClick and enchancing it in the Link component? Also, do we utilise the default link component anywhere on the JobCard? I guess it's on the demo only. In this case assigning of it should be scoped to demo only ideally and linkComponent prop should be required.

**Comment by kosanna** [comment]
Closing as will be taking more generalised approach
